src/sinol_make/commands/run/__init__.py

```diff
@@ -256,6 +256,7 @@ def perform_executions(self, compiled_commands, names, solutions, report_file):
 		program_groups_scores = collections.defaultdict(dict)
 
 		def print_view(output_file=None):
+			stdout = sys.stdout
```

Comment:
It seems you're using this variable only once in the code. Remember that each added variable increases the context a reader has to remember while trying to understand the whole code.

Reply:
I know, but if there is an output file specified the `sys.stdout` is changed to that file. And I can't simply set this value again to `sys.stdout` and the end of the function because then the `pytest`'s output capture doesn't work.

Reply:
I'm not sure I understand what you mean. Nevertheless it sounds like something that can be documented in a comment.

Reply:
Is it understandable now? 866ba0d2cf1a2b5e6e92cba382d324e548bde0fe

Reply:
Do I understand correctly that `pytest` can change any variables named `stdout` anywhere in the code for testing purposes, and you're using this mechanic with a default value `sys.stdout`? I'm still not sure how does it work and why it must be a variable.

Reply:
The default value of `sys.stdout` is `sys.__stdout__` which is a normal terminal stdout. This function sometime changes `sys.stdout` to a file. Previously, the function at the end returned `sys.stdout` to `sys.__stdout__`. But now the problem is that `pytest` changes `sys.stdout` to some custom stream to capture the output. So the easiest way to return `sys.stdout` to a correct stream at the end of the function is to have a variable that remembers what the stdout was.

Reply:
How does `sys.stdout` change? That is, why it is "not correct" at the end?

Reply:
This function can set `sys.stdout` to a file in which the executions' table will be saved. And at the end we want to return `sys.stdout` to the previous state.

Reply:
`sys.stdout = open(output_file, 'w')`. OK, I see now, thanks.

I leave it up to you to change the comment to make it more clear.

Reply:
What do you think about a comment like that:
`# sys.stdout is stored in a variable for an easy way of restoring it in case sys.stdout is changed to a file in this function`

Reply:
It's often better to start with the reason:
```
This function changes `sys.stdout` for easier printing to a file.
Because of that, we store `sys.stdout` to a variable for reverting that at the end of the function.
```

Reply:
Okay, thanks

Reply:
We can also consider using a `print` alternative that accepts a stream (file or stout) as an argument.

Reply:
There is an option in `print` to specify output stream, but then everywhere in this function we'd have to remember to specify it everywhere in this function